src/async-input.js

```diff
@@ -39,14 +39,35 @@ export class AsyncInput {
   getValue() {}
 }
 
+/**
+ * Attributes
+ *
+ * Components implementing the AsyncInput,
+ * are expected to assert these attributes
+ * at the appropriate time.
+ *
+ * @enum {string}
+ */
+export const AsyncInputAttributes = {
+  /**
+   * data-name
+   *
+   * Required attribute that must be asserted by every async-input
+   * Element. This is used by AMP form to add the key
+   * for the form submission request
+  */
+  'NAME': 'name',
```

Comment:
nit: NAME since you're using `AsyncInputAttributes.NAME` elsewhere, so that closure compiler can optimize the key string.